File: 9.java
import java.util.function.Supplier;

public class Main {
    public static void main(String[] args) {
        // Initial value of the counter
        int n = 5;

        // Create the counter function
        Supplier<Integer> counter = createCounter(n);

        // Call the counter function multiple times
        System.out.println(counter.get()); // Output: 5
        System.out.println(counter.get()); // Output: 6
        System.out.println(counter.get()); // Output: 7
    }

    // Function to create a counter
    public static Supplier<Integer> createCounter(int n) {
        return () -> n++;
    }
}

File: Typescript1/9.cpp
#include<iostream>
auto createCounter(int& n) {
    return [&n]() -> int {
        return n++;
    };
}

int main() {
    int n = 5;

    // Create the counter function
    auto counter = createCounter(n);

    // Call the counter function multiple times
    std::cout << counter() << std::endl; // Output: 5
    std::cout << counter() << std::endl; // Output: 6
    std::cout << counter() << std::endl; // Output: 7

    return 0;
}